examples/examples-log4j2/README.md

```diff
@@ -0,0 +1,62 @@
+grpc Log4j2 example
+==============================================
+
+The examples require grpc-java to already be built. You are strongly encouraged
```

Comment:
I want to see some verbiage about what this example is supposed to illustrate - possibly at the top of the file. How about adding information like (feel free to edit or add):

"This example illustrates how to properly set the ThreadContext in a server-interceptor that is then used by the logger in your server. The server-interceptor in this example sets a requestId (a randomly generated string) and the client name from the request header in the ThreadContext. The log statement in the server method will include these 2 things in the log output".